### PR TITLE
Fix language rule merge when key is unset in both rulesets

### PR DESCRIPTION
**Pull request by edg2s**
This fixes not-es2016.js where this occurs.

### CODE REVIEW

**Comment by edg2s** [comment]
We should release this as 13.1

**Comment by jdforrester** [comment]
Test failures are sort-of real. Eurgh.

File: language/merge.js [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
+	Object.assign( mergedRules.rules, parentRules.rules, childRules.rules );
 
 	// For the specified keys, concatenate the lists
 	[ 'no-restricted-syntax', 'no-restricted-properties' ].forEach( function ( key ) {
-		// If either is unset, return the other
-		if ( !parentRules.rules[ key ] ) {
+		if ( !mergedRules.rules[ key ] ) {
+			// If both are unset, do nothing
+			return;
+		} else if ( !parentRules.rules[ key ] ) {
```
Comment (Krinkle) [inline language/merge.js]: These to no-parent and no-child branches seem redundant, if I understand correctly, the below assignment results in the same outcome as the main merge above already did (a no-op). Is that right? Maybe I'm missing something...

If so, then these two branches could be removed, with the last `else` becoming an `else if` for `parent && child`. Alternatively, it looks like maybe that branch would work fine unconditionally as well, given it already has fallbacks for both of them being absent. Those fallbacks are currently inaccessible given these two earlier branches.

**Review by Krinkle — CHANGES_REQUESTED** [review]
Either the inline fallbacks or the two extra branches seem redundant.

File: language/merge.js (hunk 1 above)
Comment (edg2s) [reply language/merge.js]: Yeah - it wasn't redundant when I didn't hardcode "error". Fixed.

**Review by Krinkle — APPROVED** [review]

**Pull request merged**